wpilibc/src/main/native/cpp/event/BooleanEvent.cpp

```diff
@@ -7,7 +7,11 @@
 using namespace frc;
 
 BooleanEvent::BooleanEvent(EventLoop* loop, std::function<bool()> condition)
-    : m_loop(loop), m_condition(std::move(condition)) {}
+    : m_loop(loop), m_condition(std::move(condition)) {
+  m_loop->Bind([condition = m_condition, &state = m_state] {
+    state = std::make_shared<bool>(condition());
```

Comment:
Don't make a new pointer, it defies the whole point. Mutate the existing one. 
Also, shared_ptrs should be captured by value and not by reference.